root/edit/details/RemoveMedium.js

```diff
@@ -9,45 +9,109 @@
 
 import * as React from 'react';
 
+import {EDIT_STATUS_OPEN} from '../../constants';
 import MediumTracklist
   from '../../medium/MediumTracklist';
 import MediumLink
   from '../../static/scripts/common/components/MediumLink';
+import Warning from '../../static/scripts/common/components/Warning';
+import {reduceArtistCredit}
+  from '../../static/scripts/common/immutable-entities';
+import {arraysEqual} from '../../static/scripts/common/utility/arrays';
 
 type Props = {
   +allowNew?: boolean,
   +edit: RemoveMediumEditT,
 };
 
-const RemoveMedium = ({edit}: Props): React.Element<'table'> => {
+const areTracksEqual = (a, b) => (
+  a.name === b.name &&
+  reduceArtistCredit(a.artistCredit) === reduceArtistCredit(b.artistCredit) &&
+  a.length === b.length
+);
+
+const RemoveMedium = ({
+  edit,
+}: Props): React.Element<typeof React.Fragment> => {
   const display = edit.display_data;
+  const originalTracklist = display.tracks ?? [];
+  const currentTracklist = display.medium.tracks ?? [];
+  let showTracklistAndWarning = false;
+  let hasLengthChanges = false;
+  if (edit.status === EDIT_STATUS_OPEN) {
+    hasLengthChanges =
+      originalTracklist.length !== currentTracklist.length;
+    const areTracklistsEqual = arraysEqual(
+      originalTracklist,
+      currentTracklist,
+      areTracksEqual,
+    );
+    showTracklistAndWarning = !areTracklistsEqual;
+  }
 
   return (
-    <table className="details remove-medium">
-      <tr>
-        <th>{addColonText(l('Medium'))}</th>
-        <td>
-          <MediumLink medium={display.medium} />
-        </td>
-      </tr>
-
-      <tr>
-        <th>{addColonText(l('Tracklist'))}</th>
-        <td>
-          <table className="tbl">
-            <tbody>
-              {display.tracks?.length ? (
-                <MediumTracklist
-                  showArtists
-                  tracks={display.tracks}
-                />
-              ) : l('The tracklist for this medium is unknown.')}
-            </tbody>
-          </table>
-        </td>
-      </tr>
-
-    </table>
+    <>
+      {showTracklistAndWarning ? (
+        <Warning
+          message={hasLengthChanges
+            ? l(`The number of tracks on the medium being removed
+                 has changed since the removal edit was entered.
+                 Please check the changes and ensure the removal
+                 is still correct.`)
+            : l(`Some track lengths, titles or artists have changed
+                 since the removal edit was entered. Please check the changes
+                 and ensure the removal is still correct.`)
```

Comment:
It would be cool to show the changes as an EditMedium type diff so that multiple messages aren't needed saying what to look for, but not sure how feasible that is.

Reply:
It's probably doable, but the calculation of tracklist_changes is not trivial, so I think I'll merge like this and add a ticket for possibly adapting it later.

Reply:
Added MBS-12236 for that